packages/react-component-manifests/src/manifests/Carousel/Carousel.tsx

```diff
@@ -1,255 +1,61 @@
-                        border: "none",
-                        width: "40px",
-                        position: "relative",
-                        zIndex: "1",
-                        height: "1px",
-                        transition: "width 0.3s ease-in",
-                        color: "#fff",
-                      }
-                }
-                onClick={() => {
-                  updateIndex(index);
-                }}
-              ></button>
-            );
-          })}
-        </div>
-      )}
-      {isCircle && (
-        <div style={{ display: "flex", justifyContent: "center" }}>
-          {React.Children.map(children, (child, index) => {
-            return (
-              <button
-                style={
-                  index === activeIndex
-                    ? {
-                        margin: "5px",
-                        border: "none",
-                        width: "12px",
-                        marginTop: "-15px",
-                        position: "relative",
-                        zIndex: "1",
-                        height: "12px",
-                        transition: "width 0.3s ease-in",
-                        borderRadius: "50%",
-                        backgroundColor: "#222",
-                      }
-                    : {
-                        margin: "5px",
-                        border: "none",
-                        width: "12px",
-                        marginTop: "-15px",
-                        position: "relative",
-                        zIndex: "1",
-                        height: "12px",
-                        transition: "width 0.3s ease-in",
-                        borderRadius: "50%",
-                        backgroundColor: "#fff",
-                      }
-                }
-                onClick={() => {
-                  updateIndex(index);
-                }}
-              ></button>
-            );
-          })}
-        </div>
-      )}
-    </div>
-  );
-};
+export type CarouselEffect = "scrollx" | "fade";
+export type DotPosition = "top" | "bottom" | "left" | "right";
 
 const Carousel = forwardRef<
   HTMLDivElement,
   {
     styles: React.CSSProperties;
     custom: {
       items: { text: string; image?: string }[];
-      startTile: number;
-      isIndicatorCircle: boolean;
-      indicatorPosition: string;
+      effect?: CarouselEffect;
+      dotPosition?: DotPosition;
+      dots?:
+        | boolean
+        | {
+            className?: string;
+          };
+      beforeChange?(currentSlide: number, nextSlide: number): void;
     };
     onClick: (event: { pageX: number; pageY: number }) => void;
+
     className?: string;
   }
 >((props, ref) => {
-  const onClick = useCallback(
-    (e: React.MouseEvent) => {
-      props.onClick({ pageX: e.pageX, pageY: e.pageY });
-    },
-    [props]
-  );
+  const { custom } = props;
+  const onChange = (currentSlide: number) => {
+    console.log(currentSlide);
```

Comment:
I think you should attach an onChange callback to the manifest of Carousel and call that here.